src/main/java/com/robinpowered/sdk/RobinApi.java

```diff
@@ -134,6 +135,16 @@ public boolean hasCredential() {
     }
 
     /**
+     * Returns whether a the interface is authenticated with a valid credential.
+     * Valid credentials have non-null values.
+     *
+     * @return Whether a valid credential has been set.
+     */
+    public boolean isAuthenticated() {
+        return credential != null && credential.getValue() != null;
```

Comment:
Maybe we already mentioned this, but why not just `getCredential` and have client check the value? It's kind of their responsibility to begin with to manage the details of the credential.